### PR TITLE
Require Numba 0.57.0+ & NumPy 1.21.0+

### PR DESCRIPTION
**Pull request by jakirkham**
Align with the rest of RAPIDS on these requirements. Also needed for CUDA 12 support.

### CODE REVIEW

**Comment by jakirkham** [comment]
Could someone please add the "improvement" & "non-breaking" labels?

File: conda/recipes/cuxfilter/meta.yaml [hunk 1]
```diff
@@ -34,7 +34,8 @@ requirements:
     - jupyter-server-proxy
     - libwebp
     - nodejs >=14
-    - numba >=0.54
+    - numba >=0.57
+    - numba >=1.21
```
Comment (bdice) [inline conda/recipes/cuxfilter/meta.yaml:38]:
```suggestion
    - numpy >=1.21
```

Comment (jakirkham) [reply conda/recipes/cuxfilter/meta.yaml]: Oops 🤦‍♂️

Should now be fixed 😅

**Comment by AjayThorve** [comment]
The docs build is failing, due to an issue where numpy v1.24.3 prevent holoviews to render cuxfilter docs charts, should be fixed in #479 [here](https://github.com/AjayThorve/cuxfilter/blob/fix/failing-tests/dependencies.yaml#L76)

**Comment by bdice** [comment]
> The docs build is failing, due to an issue where numpy v1.24.3 prevent holoviews to render cuxfilter docs charts

@AjayThorve Is there an upstream bug in holoviews / numpy that we can link to here so that we know when this limitation is lifted? This is important to track for all of RAPIDS, because cuxfilter is installed by default in the `rapids` metapackage and users may have environment conflicts that require NumPy 1.24. We've seen NumPy 1.24 compatibility requests for cudf and were able to lift this upper bound.

**Comment by AjayThorve** [comment]
> @AjayThorve Is there an upstream bug in holoviews / numpy that we can link to in that PR so that we know when this limitation is lifted? This is important to track for all of RAPIDS, because `cuxfilter` is installed by default in the `rapids` metapackage and users may have environment conflicts that require NumPy 1.24. We've seen NumPy 1.24 compatibility requests for cudf and were able to lift this upper bound.

Hey @bdice, so this issue is already fixed in datashader (https://github.com/holoviz/datashader/pull/1176), although it's not yet part of any release(the latest released datashader version is 0.14.4, and this PR is part of the 0.14.5 milestone). How do you suggest we proceed in this scenario?

**Comment by jakirkham** [comment]
Can we raise an issue on their repo requesting a release (and link that issue here)?

**Comment by AjayThorve** [comment]
Have asked for an ETA for release here (https://discourse.holoviz.org/t/module-numpy-has-no-attribute-warnings/5402)

**Comment by Hoxbro** [comment]
We have released datashader 1.15, and are working on adding it to conda-forge [here](https://github.com/conda-forge/datashader-feedstock/pull/28). So hopefully you will soon be unblocked.

**Comment by ianthomas23** [comment]
Conda-forge datashader-feedstock v0.15.0 PR has been merged so you should be able to proceed shortly.

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Review by raydouglass — APPROVED** [review]

**Comment by AjayThorve** [comment]
thanks everyone, #479 should fix the CI issues, and this PR should be good to merge after that

**Comment by AjayThorve** [comment]
@jakirkham can you merge/rebase with `main`, that should fix the CI failures

**Review by AjayThorve — APPROVED** [review]

**Comment by jakirkham** [comment]
Toggling for CI

**Review by AjayThorve — APPROVED** [review]

**Review by bdice — CHANGES_REQUESTED** [review]
Please update `dependencies.yaml` to align. We need `numpy` and `numba` pinnings in the `run` section: https://github.com/rapidsai/cuxfilter/blob/472f839e5eb9864dbc772e414b6759d62505f2e8/dependencies.yaml#L107-L128

**Review by raydouglass — APPROVED** [review]

**Comment by AjayThorve** [comment]
> Please update `dependencies.yaml` to align. We need `numpy` and `numba` pinnings and to make the `run` section align with the conda list of run dependencies:

@bdice, I pushed the requested changes

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**